app-android/src/main/kotlin/com/ustadmobile/port/android/view/RoleEditFragment.kt

```diff
@@ -0,0 +1,168 @@
+        }
+
+    private val roleObserver = Observer<List<BitmaskFlag>?>{
+        t -> mRecyclerViewAdapter?.submitList(t)
+    }
+
+
+    class BitmaskViewHolder(val itemBinding: ItemBitmaskBinding)
+        : RecyclerView.ViewHolder(itemBinding.root)
+
+    class BitmaskRecyclerViewAdapter()
+        : ListAdapter<BitmaskFlag, BitmaskViewHolder>(DIFFUTIL_BITMASKFLAG) {
+
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BitmaskViewHolder {
+            val viewHolder =  BitmaskViewHolder(ItemBitmaskBinding.inflate(
+                    LayoutInflater.from(parent.context),
+                    parent, false))
+            return viewHolder
+        }
+
+        override fun onBindViewHolder(holder: BitmaskViewHolder, position: Int) {
+            holder.itemBinding.bitmaskFlag = getItem(position)
+        }
+    }
+
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val rootView : View
+        mBinding = ActivityRoleEditBinding.inflate(inflater, container, false).also {
+            rootView = it.root
+            mRecyclerView = it.roleEditPermissionBitmaskEditRv
+        }
+
+        mPresenter = RoleEditPresenter(requireContext(), arguments.toStringMap(), this,
+                this, UstadMobileSystemImpl.instance,
+                UmAccountManager.getActiveDatabase(requireContext()),
+                UmAccountManager.getRepositoryForActiveAccount(requireContext()),
+                UmAccountManager.activeAccountLiveData)
+        //mPresenter?.onCreate(savedInstanceState.toNullableStringMap())
+
+        mRecyclerViewAdapter = BitmaskRecyclerViewAdapter()
+        mRecyclerView?.adapter = mRecyclerViewAdapter
+        mRecyclerView?.layoutManager = LinearLayoutManager(requireContext())
+
+
+
+        return rootView
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        mPresenter?.onCreate(savedInstanceState.toNullableStringMap())
+
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        mBinding = null
+        mPresenter = null
+        entity = null
+    }
+
+    override var fieldsEnabled: Boolean = false
+        get() = field
+        set(value) {
+            field = value
+            mBinding?.fieldsEnabled = value
+        }
+
+    override var loading: Boolean = false
+        get() = field
+        set(value) {
+            field = value
+            mBinding?.loading = value
+        }
+
+    override fun onResume() {
+        super.onResume()
+        setEditFragmentTitle(R.string.role)
+    }
+
+
+    companion object {
+
+        val DIFFUTIL_BITMASKFLAG = object: DiffUtil.ItemCallback<BitmaskFlag>() {
+            override fun areItemsTheSame(oldItem: BitmaskFlag, newItem: BitmaskFlag): Boolean {
+                return oldItem.flagVal == newItem.flagVal
+            }
+
```

Comment:
I think we should add a comment here about the === being required to ensure that the two-way binding saves the data to the latest reference of the object.